document/src/vespa/document/base/field.cpp

```diff
@@ -7,9 +7,21 @@
 #include <vespa/vespalib/util/exceptions.h>
 #include <vespa/vespalib/stllike/asciistream.h>
 #include <vespa/vespalib/util/bobhash.h>
+#include <algorithm>
 
 namespace document {
 
+Field::Set::Set(std::vector<CPtr> fields)
+    : _fields(std::move(fields))
+{
+    std::sort(_fields.begin(), _fields.end(), Field::FieldPtrComparator());
```

Comment:
consider doing unique as well, to ensure set property

Reply:
Yes, thanks.